datadog_lambda/tracing.py

```diff
@@ -492,67 +477,47 @@ def extract_dd_trace_context(
     event_source = parse_event_source(event)
+    print(f"Debug: Event source determined as {event_source}")  # Debug statement
 
     if extractor is not None:
-        (
-            trace_id,
-            parent_id,
-            sampling_priority,
-        ) = extract_context_custom_extractor(extractor, event, lambda_context)
+        print("Debug: Using custom extractor")  # Debug statement
+        context = extract_context_custom_extractor(extractor, event, lambda_context)
     elif isinstance(event, (set, dict)) and "headers" in event:
-        (
-            trace_id,
-            parent_id,
-            sampling_priority,
-        ) = extract_context_from_http_event_or_context(
+        print("Debug: Extracting context from HTTP event or context")  # Debug statement
+        context = extract_context_from_http_event_or_context(
             event, lambda_context, event_source, decode_authorizer_context
         )
     elif event_source.equals(EventTypes.SNS) or event_source.equals(EventTypes.SQS):
-        (
-            trace_id,
-            parent_id,
-            sampling_priority,
-        ) = extract_context_from_sqs_or_sns_event_or_context(event, lambda_context)
+        print("Debug: Extracting context from SNS or SQS event or context")  # Debug statement
+        context = extract_context_from_sqs_or_sns_event_or_context(event, lambda_context)
     elif event_source.equals(EventTypes.EVENTBRIDGE):
-        (
-            trace_id,
-            parent_id,
-            sampling_priority,
-        ) = extract_context_from_eventbridge_event(event, lambda_context)
+        print("Debug: Extracting context from EventBridge event")  # Debug statement
+        context = extract_context_from_eventbridge_event(event, lambda_context)
     elif event_source.equals(EventTypes.KINESIS):
-        (
-            trace_id,
-            parent_id,
-            sampling_priority,
-        ) = extract_context_from_kinesis_event(event, lambda_context)
+        print("Debug: Extracting context from Kinesis event")  # Debug statement
+        context = extract_context_from_kinesis_event(event, lambda_context)
     elif event_source.equals(EventTypes.STEPFUNCTIONS):
-        (
-            trace_id,
-            parent_id,
-            sampling_priority,
-        ) = extract_context_from_step_functions(event, lambda_context)
+        print("Debug: Extracting context from Step Functions")  # Debug statement
+        context = extract_context_from_step_functions(event, lambda_context)
     else:
-        trace_id, parent_id, sampling_priority = extract_context_from_lambda_context(
-            lambda_context
-        )
+        print("Debug: Defaulting to extract context from Lambda context")  # Debug statement
+        context = extract_context_from_lambda_context(lambda_context)
 
-    if trace_id and parent_id and sampling_priority:
+    if context is not None and context.trace_id and context.span_id and context.sampling_priority:
```

Comment:
I don't think you need to check for trace id span id and priority. If one of them was not found in the headers, then the propagator.extract will return None.